Libraries/Text/RCTTextField.m

```diff
@@ -112,6 +113,85 @@ - (void)setPlaceholder:(NSString *)placeholder
   RCTUpdatePlaceholder(self);
 }
 
+- (UIColor *)tintColor
+{
+  return super.tintColor;
+}
+
+- (void)setTintColor:(UIColor *)tintColor
+{
+  super.tintColor = tintColor;
+}
+
+- (UIColor *)clearButtonTintColor
+{
+  return _clearButtonTintColor;
+}
+
+- (void)setClearButtonTintColor:(UIColor *)clearButtonTintColor
+{
+    _clearButtonTintColor = clearButtonTintColor;
+}
+
+static UIButton* findClearButton(RCTTextField *self)
+{
+    for(UIView *v in self.subviews)
+    {
+        if([v isKindOfClass:[UIButton class]])
+        {
+            UIButton *buttonClear = (UIButton *) v;
+            return buttonClear;
+        }
+    }
+    return nil;
+}
+
+-(void) layoutSubviews
+{
+    [super layoutSubviews];
+    tintClearButton(self);
+}
+
+static void tintClearButton(RCTTextField *self)
+{
+    UIButton *clearButton = findClearButton(self);
+
+    if(self.clearButtonTintColor && clearButton)
+    {
+        UIImage *imageNormal = [clearButton imageForState:UIControlStateNormal];
+        UIImage *imageHighlighted = [clearButton imageForState:UIControlStateHighlighted];
+
+        UIImage *tintedImageNormal = tintedImage(imageNormal, self.clearButtonTintColor);
```

Comment:
Instead of drawing a new image, it might be easier to create a new UIImage with renderingMode = `UIImageRenderingModeAlwaysTemplate` and set the tintColor of the UIButton.